### PR TITLE
add lint and phpunit tests to drone

### PR DESCRIPTION
**Pull request by individual-it**
leaving tests on travis running for now in parallel

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -117,9 +123,36 @@ pipeline:
...
+    environment:
+      - PHP_VERSION=${PHP_VERSION}
+    commands:
+      - cd /var/www/owncloud
+      - chmod +x lib/composer/phpunit/phpunit/phpunit
+      - rm lib/composer/bin/phpunit
+      - ln -s ../phpunit/phpunit/phpunit lib/composer/bin/phpunit
+      - phpdbg -d memory_limit=4096M -rr ./lib/composer/bin/phpunit --configuration apps/user_ldap/phpunit.xml
+      - cd apps/user_ldap/
+      - wget -q https://scrutinizer-ci.com/ocular.phar
```
Comment (patrickjahns) [inline .drone.yml]: lets replace this with codecov

File: .drone.yml [hunk 2]
```diff
@@ -117,9 +123,36 @@ pipeline:
       matrix:
         TEST_SUITE: integration
 
+  list-test:
```
Comment (patrickjahns) [inline .drone.yml]: `lint` ?

**Comment by patrickjahns** [comment]
Small Typo in `NEED_INSTAL_APP`

File: .drone.yml [hunk 3]
```diff
@@ -117,9 +135,34 @@ pipeline:
...
+    when:
+      matrix:
+        TEST_SUITE: lint
+
+  phpunit-tests:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - cd /var/www/owncloud
+      - chmod +x lib/composer/phpunit/phpunit/phpunit
```
Comment (DeepDiver1975) [inline .drone.yml]: what is this? we better fix the release tar ball generation before spreading this workaround trough out to all repos.

File: .drone.yml [hunk 4]
```diff
@@ -117,9 +135,34 @@ pipeline:
...
+      - rm lib/composer/bin/phpunit
+      - ln -s ../phpunit/phpunit/phpunit lib/composer/bin/phpunit
+      - phpdbg -d memory_limit=4096M -rr ./lib/composer/bin/phpunit --configuration apps/user_ldap/phpunit.xml
+      - cd apps/user_ldap/
+      - wget -qO codecov.sh https://codecov.io/bash
```
Comment (DeepDiver1975) [inline .drone.yml]: use the existing drone plugin please

File: .drone.yml [hunk 5]
```diff
@@ -117,9 +135,34 @@ pipeline:
       matrix:
         TEST_SUITE: integration
 
+  lint-test:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - find . -name \*.php -not -path './vendor/*' -exec php -l "{}" \;
```
Comment (DeepDiver1975) [inline .drone.yml]: please use this tool instead - https://github.com/JakubOnderka/PHP-Parallel-Lint

File: .drone.yml (hunk 3 above)
Comment (individual-it) [reply .drone.yml]: got the idea from you ;-)
https://github.com/owncloud/administration/pull/124/commits/1e7a058d66f21dbe4a83da8f75e9d7b5eca996f1

Comment (butonic) [reply .drone.yml]: we could use `php lib/composer/bin/phpunit`? no need to make it executeable and copy it around?

**Comment by butonic** [comment]
I am happy to merge this. I need it for https://github.com/owncloud/user_ldap/pull/199

**Comment by individual-it** [comment]
@butonic yes I'm also happy to merge it, need it also for #183
But waiting for https://github.com/owncloud/core/pull/30678 & https://github.com/owncloud/administration-internal/pull/96 to be merged so can revert that hack @DeepDiver1975  commented on.
Happy to merge it (if someone approved) and make a new issue & PR to get that hack out again

**Comment by individual-it** [comment]
issue to fix the workaround here https://github.com/owncloud/user_ldap/issues/201

**Pull request merged**